pkg/runtime/framework/plugins/jobset/jobset_test.go

```diff
@@ -0,0 +1,331 @@
+/*
+Copyright 2025 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobset
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	batchv1ac "k8s.io/client-go/applyconfigurations/batch/v1"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/klog/v2/ktesting"
+	"k8s.io/utils/ptr"
+	jobsetv1alpha2ac "sigs.k8s.io/jobset/client-go/applyconfiguration/jobset/v1alpha2"
+
+	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
+	"github.com/kubeflow/trainer/pkg/constants"
+	"github.com/kubeflow/trainer/pkg/runtime"
+	"github.com/kubeflow/trainer/pkg/runtime/framework"
+	utiltesting "github.com/kubeflow/trainer/pkg/util/testing"
+)
+
+// TODO: Add tests for all Interfaces.
+// REF: https://github.com/kubeflow/trainer/issues/2468
+
+func TestJobSet(t *testing.T) {
```

Comment:
Do we have any duplicated tests in Plugins and TrainingRuntime ?
https://github.com/kubeflow/trainer/blob/8282e748c78b700d0682f91778daeb1bed66f6d2/pkg/runtime/core/trainingruntime_test.go

Reply:
Duplicated tests will be removed as part of https://github.com/kubeflow/trainer/issues/2468 in the future PR.